confmap/internal/e2e/expand_test.go

```diff
@@ -5,90 +5,99 @@ package e2etest
 
-	expectedFailures := map[string]string{
-		"$ENV_VALUE":   "variable substitution using $VAR has been deprecated in favor of ${VAR} and ${env:VAR}",
-		"$$$ENV_VALUE": "variable substitution using $VAR has been deprecated in favor of ${VAR} and ${env:VAR}",
+	expectedMap := map[string]any{
+		"test_map": map[string]any{
+			"key1":  "$ENV_VALUE",
+			"key2":  "$$ENV_VALUE",
+			"key3":  "$${ENV_VALUE}",
+			"key4":  "some${ENV_VALUE}text",
+			"key5":  "some${ENV_VALUE}text",
+			"key6":  "${ONE}${TWO}",
+			"key7":  "text$",
+			"key8":  "$",
+			"key9":  "${1}${env:2}",
+			"key10": "some${env:ENV_VALUE}text",
+			"key11": "${env:${ENV_VALUE}}",
+			"key12": "${env:${ENV_VALUE}}",
+			"key13": "env:MAP_VALUE_2}${ENV_VALUE}{",
+			"key14": "$" + expandedValue,
+			"key15": "$ENV_VALUE",
+			"key16": []any{"$ESCAPE_ME", "${ESCAPE_ME}", "${env:ESCAPE_ME}"},
+			"key17": map[string]any{"key1": "$ESCAPE_ME", "key2": "${ESCAPE_ME}", "key3": "${env:ESCAPE_ME}"},
+		},
 	}
 
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			expectedMap := map[string]any{
-				"test_map": map[string]any{
-					"key1":  "$ENV_VALUE",
-					"key2":  "$$ENV_VALUE",
-					"key3":  "$" + expandedValue,
-					"key4":  "some" + expandedValue + "text",
-					"key5":  "some${ENV_VALUE}text",
-					"key6":  "${ONE}${TWO}",
-					"key7":  "text$",
-					"key8":  "$",
-					"key9":  "${1}${env:2}",
-					"key10": "some${env:ENV_VALUE}text",
-					"key11": "${env:" + expandedValue + "}",
-					"key12": "${env:${ENV_VALUE}}",
-					"key13": "env:MAP_VALUE_2}${ENV_VALUE}{",
-					"key14": "$" + expandedValue,
-				},
-			}
+	resolver, err := confmap.NewResolver(confmap.ResolverSettings{
+		URIs:              []string{filepath.Join("testdata", "expand-escaped-env.yaml")},
+		ProviderFactories: []confmap.ProviderFactory{fileprovider.NewFactory(), envprovider.NewFactory()},
+		DefaultScheme:     "",
+	})
+	require.NoError(t, err)
 
-			resolver, err := confmap.NewResolver(confmap.ResolverSettings{
-				URIs:               []string{filepath.Join("testdata", "expand-escaped-env.yaml")},
-				ProviderFactories:  []confmap.ProviderFactory{fileprovider.NewFactory(), envprovider.NewFactory()},
-				ConverterFactories: []confmap.ConverterFactory{expandconverter.NewFactory()},
-				DefaultScheme:      tt.scheme,
-			})
-			require.NoError(t, err)
+	// Test that expanded configs are the same with the simple config with no env vars.
+	cfgMap, err := resolver.Resolve(context.Background())
+	require.NoError(t, err)
+	m := cfgMap.ToStringMap()
+	assert.Equal(t, expectedMap, m)
+}
 
-			// Test that expanded configs are the same with the simple config with no env vars.
-			cfgMap, err := resolver.Resolve(context.Background())
-			require.NoError(t, err)
-			m := cfgMap.ToStringMap()
-			assert.Equal(t, expectedMap, m)
+// Test_EscapedEnvVars tests that the resolver supports escaped env vars working together with expand converter.
```

Comment:
```suggestion
```
I think the function name should cover this, especially now that the expand converter is no longer involved.